src/select/mini.rs

```diff
@@ -460,29 +485,50 @@ where
     }
 
     fn view_direction_item(ctx: &Context<Self>) -> Html {
-        let txt = ctx.props().txt.txt.clone();
-        let value = if let Ok(selected) = ctx.props().selected_value.try_borrow() {
-            selected.map_or_else(String::new, |value| {
-                ctx.props()
-                    .candidate_values
-                    .iter()
-                    .enumerate()
-                    .find(|(_, v)| *v == &value)
-                    .map_or_else(String::new, |(index, _)| {
-                        ctx.props()
-                            .list
-                            .get(index)
-                            .map_or_else(String::new, |v| match v {
-                                ViewString::Key(key) => {
-                                    text!(txt, ctx.props().language, key).to_string()
-                                }
-                                ViewString::Raw(txt) => txt.clone(),
-                            })
-                    })
-            })
+        let selected_value = ctx
+            .props()
+            .selected_value
+            .try_borrow()
+            .ok()
+            .and_then(|selected| selected.as_ref().copied());
+        let default_value = if ctx.props().active {
+            None
+        } else {
+            ctx.props().default_value
+        };
+        let first_candidate = if ctx.props().active {
+            None
+        } else {
+            ctx.props().candidate_values.first().copied()
+        };
+        let value = [selected_value, default_value, first_candidate]
+            .into_iter()
+            .flatten()
+            .find_map(|candidate| Self::value_to_text(ctx, &candidate))
+            .unwrap_or_default();
+        let suffix = if ctx.props().active {
+            String::new()
+        } else if let Some(suffix) = ctx.props().deactive_class_suffix.as_ref() {
+            suffix.as_ref().into()
         } else {
             String::new()
         };
+        let class_base = format!("mini-select-item-direction{suffix}");
```

Comment:
Please minimize the use of `clone`. Consider using `as_str` or applying modifications through `map` instead.